modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/MuleLoggerContext.java

```diff
@@ -56,22 +63,44 @@
   MuleLoggerContext(String name, URI configLocn, ClassLoader ownerClassLoader, ContextSelector contextSelector,
                     boolean standalone) {
     super(name, null, configLocn);
-    configFile = configLocn;
+    this.configFile = configLocn;
     this.contextSelector = contextSelector;
     this.standlone = standalone;
-    ownerClassLoaderHash =
+    this.ownerClassLoaderHash =
         ownerClassLoader != null ? ownerClassLoader.hashCode() : getClass().getClassLoader().getSystemClassLoader().hashCode();
 
-    if (ownerClassLoader instanceof ArtifactClassLoader) {
-      artifactClassloader = true;
-      artifactName = getArtifactName((ArtifactClassLoader) ownerClassLoader);
-      artifactDescriptor = getArtifactDescriptor((ArtifactClassLoader) ownerClassLoader);
-      applicationClassloader = ownerClassLoader instanceof RegionClassLoader;
+    Optional<ArtifactClassLoader> artifactClassloader = getArtifactClassLoader(ownerClassLoader);
```

Comment:
how exactly does this achieve the goal? As far as I can see, this only properly obtains artifact metadata, but the actual logging separation is done in the `DispatchingLogger` class through the `ownerClassLoaderHash` field , which is not changed here.

Also, I see you added tests about the artifact metadata being properly obtained, and that's fine. But I see no test verifying that the policy log is actually going where you want.

Reply:
We achieve this by changing the artifactName that the context uses.
Let me check how can I add those tests.